src/java.base/share/classes/java/security/Provider.java

```diff
@@ -1557,12 +1557,22 @@ private static class EngineDescription {
         final String name;
         final boolean supportsParameter;
         final String constructorParameterClassName;
+        private volatile Class<?> constructorParameterClass;
```

Comment:
Style: no need for `private` here, match what other fields are doing.

Reply:
I don't disagree in principle but it was like this before the revert, and is still like this in 17.

Reply:
Is volatile really needed? And there is some performance penalty and in practice the value will be the same even if recomputed.

Reply:
Same thinking as above - this is how it was before, and how it is in 17. I'd rather not diverge, unless the reason is strong.